app/src/main/java/com/github/campus_capture/bootcamp/fragments/RegisterFragment.java

```diff
@@ -0,0 +1,199 @@
+package com.github.campus_capture.bootcamp.fragments;
+
+import static android.content.ContentValues.TAG;
+
+import android.app.AlertDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+
+import com.github.campus_capture.bootcamp.AppContext;
+import com.github.campus_capture.bootcamp.R;
+import com.github.campus_capture.bootcamp.activities.MainActivity;
+import com.github.campus_capture.bootcamp.authentication.Section;
+import com.github.campus_capture.bootcamp.authentication.TOS;
+import com.github.campus_capture.bootcamp.authentication.User;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class RegisterFragment extends Fragment {
+
+
+    private Button already_registered_button;
+    private Button register_button;
+    private Button spectator_button;
+    private EditText email;
+    private String emailText;
+    private EditText password;
+    private String passwordText;
+    private FirebaseAuth mAuth;
+
+
+    public RegisterFragment() {
+        // Required empty public constructor
+    }
+
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view = inflater.inflate(R.layout.fragment_register, container, false);
+
+        // Init buttons
+        already_registered_button = view.findViewById(R.id.register_already_registered_button);
+        register_button = view.findViewById(R.id.register_button);
+        spectator_button = view.findViewById(R.id.register_spectator_button);
+
+        // Init texts
+        email = view.findViewById(R.id.register_email_address);
+        password = view.findViewById(R.id.register_password);
+
+        // Init Auth (Authenticater)
+        AppContext context = AppContext.getAppContext();
+        mAuth = context.getFirebaseAuth();
+
+        // Init listeners on the buttons
+        setAlreadyRegisteredButtonListener();
+        setRegisterButtonListener();
+        setSpectatorButtonListener();
+        // Inflate the layout for this fragment
+        return view;
+    }
+
+    private void setRegisterButtonListener(){
+        register_button.setOnClickListener(view -> registerClicked());
+    }
```

Comment:
don't need a function : (at 79) `register_button.setOnClickListner(this::registerClicked)`

Reply:
But I think it modularizes the code and so makes it easier to read. Isn't it?